logger/modules/logger.py

```diff
@@ -0,0 +1,152 @@
+"""
+Logs debug messages.
+"""
+
+import datetime
+import inspect
+import logging
+import pathlib
+import os
+
+# Used in type annotation of logger parameters
+# pylint: disable-next=unused-import
+import types
+
+from read_yaml.modules import read_yaml
+
+
+CONFIG_FILE_PATH = pathlib.Path("config_logger.yaml")
+
+
+class Logger:
+    """
+    Instantiates Logger objects.
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(cls, name: str, enable_log_to_file: bool) -> "tuple[bool, Logger | None]":
+        """
+        Create and configure a logger.
+        """
+        # Configuration settings
+        result, config = read_yaml.open_config(CONFIG_FILE_PATH)
+        if not result:
+            print("ERROR: Failed to load configuration file")
+            return False, None
+
+        # Get Pylance to stop complaining
+        assert config is not None
+
+        try:
+            log_directory_path = config["logger"]["directory_path"]
+            file_datetime_format = config["logger"]["file_datetime_format"]
+            logger_format = config["logger"]["format"]
+            logger_datetime_format = config["logger"]["datetime_format"]
+        except KeyError as exception:
+            print(f"Config key(s) not found: {exception}")
+            return False, None
+
+        # Create a unique logger instance
+        logger = logging.getLogger(name)
+        logger.setLevel(logging.DEBUG)
+
+        formatter = logging.Formatter(
+            fmt=logger_format,
+            datefmt=logger_datetime_format,
+        )
+
+        # Handles logging to terminal
+        stream_handler = logging.StreamHandler()
+        stream_handler.setFormatter(formatter)
+        logger.addHandler(stream_handler)
+
+        # Handles logging to file
+        if enable_log_to_file:
+            # Get the path to the logs directory.
+            entries = os.listdir(log_directory_path)
+
+            if len(entries) == 0:
+                print("ERROR: Must create a new log directory for this run before starting logger")
```

Comment:
Rather than "log directory", which is confusing with the base "directory_path" from the yaml, we can say "No previous logging sessions available"